pkg/deploy/envParams.go

```diff
@@ -30,18 +54,8 @@ func ApplyParams(componentPath string, imageParamsMap map[string]string, isUpdat
 		return err
 	}
 
-	defer paramsEnv.Close()
-
-	paramsEnvMap := make(map[string]string)
-	scanner := bufio.NewScanner(paramsEnv)
-	for scanner.Scan() {
-		line := scanner.Text()
-		parts := strings.SplitN(line, "=", 2)
-		if len(parts) == 2 {
-			paramsEnvMap[parts[0]] = parts[1]
-		}
-	}
-	if err := scanner.Err(); err != nil {
+	tmp, err := os.CreateTemp(componentPath, "params.env-")
+	if err != nil {
 		return err
 	}
 
```

Comment:
- to add a defer in case during `fmt.Fprintf `or `write.Flush()` any error to keep temp file open
- use /tmp to dump these temp parms.env-XXXX file. unless we have to use the same path "componentPath"?
```suggestion
	tmp, err := os.CreateTemp("/tmp", "params.env-")
	if err != nil {
		return err
	}
	defer tmp.Close()
```

Reply:
- valid, error paths do not close it. I'll refactor it moving to an own function like reading to have only close in defer. In general since it should be closed before end of ApplyParams I wanted to avoid double close since it's a error.
- os.Rename() is not such universal as "mv" command, it works only on one filesystem. For me with /tmp mounted on tmpfs it failed. That's why I replaced ""  (use default temp directory) with componentPath to be on the safe side.